src/Row.php

```diff
@@ -82,7 +84,11 @@ public function toArray($nullValue = null, $calculateFormulas = false, $formatDa
             $value = (new Cell($cell))->getValue($nullValue, $calculateFormulas, $formatData);
 
             if (isset($this->headingRow[$i])) {
-                $cells[$this->headingRow[$i]] = $value;
+                if (!$this->headerIsGrouped[$i]) {
```

Comment:
This should be an isset check as well no?

I think we can also un-nest this if statement:

```
if (isset($this->headerIsGrouped[$i])) {

elseif (isset($this->headingRow[$i])) {

else
```

Reply:
I don't think `isset()` will work on `$headerIsGrouped` because both `true` and `false` will return `true` for that header index. 

`$headerIsGrouped` is an array of booleans with same indexes as `$headingRow` and indicates whether that header should be grouped.